app/routers/sessions.py

```diff
@@ -212,15 +204,14 @@ async def update_session(session_id: str, session_updates: UpdateSession):
     ):
         # if `time_remaining` key is not present =>
         # no time limit is set, no need to respond with time_remaining
-        session["time_remaining"] = max(0, session["time_remaining"] - time_elapsed)
+        time_remaining = max(0, session["time_remaining"] - time_elapsed)
+        session_update_query["$set"] = {"time_remaining": time_remaining}
         response_content = {"time_remaining": session["time_remaining"]}
```

Comment:
should be `response_content = {"time_remaining": time_remaining]}` since we haven't updated in session yet